src/sql/workbench/contrib/executionPlan/browser/widgets/findExpensiveNodeWidget.ts

```diff
@@ -0,0 +1,310 @@
+		operationLabel.innerText = localize('expensiveOperationLabel', 'Metric:');
+
+		this._operationNameSelectBoxContainer.appendChild(operationLabel);
+		this.container.appendChild(this._operationNameSelectBoxContainer);
+
+		this._operationNameSelectBox = new SelectBox([
+			OFF_STRING,
+			ACTUAL_ELAPSED_TIME_STRING,
+			ACTUAL_ELAPSED_CPU_TIME_STRING,
+			COST_STRING,
+			SUBTREE_COST_STRING,
+			ACTUAL_NUMBER_OF_ROWS_FOR_ALL_EXECUTIONS_STRING,
+			NUMBER_OF_ROWS_READ_STRING
+		], COST_STRING, this.contextViewService, this._operationNameSelectBoxContainer);
+
+		this._operationNameSelectBox.render(this._operationNameSelectBoxContainer);
+		this._register(attachSelectBoxStyler(this._operationNameSelectBox, this.themeService));
+
+		this._operationNameSelectBoxContainer.style.width = '200px';
+		this._operationNameSelectBoxContainer.style.marginRight = '5px';
+
+		this._register(this._operationNameSelectBox.onDidSelect(e => {
+			switch (e.selected) {
+				case ACTUAL_ELAPSED_TIME_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.ActualElapsedTime;
+					break;
+				case ACTUAL_ELAPSED_CPU_TIME_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.ActualElapsedCpuTime;
+					break;
+				case COST_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.Cost;
+					break;
+				case SUBTREE_COST_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.SubtreeCost;
+					break;
+				case ACTUAL_NUMBER_OF_ROWS_FOR_ALL_EXECUTIONS_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.ActualNumberOfRowsForAllExecutions;
+					break;
+				case NUMBER_OF_ROWS_READ_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.NumberOfRowsRead;
+					break;
+				default:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.Off;
+			}
+		}));
+
+		// Apply Button
+		const findExpensiveOperationAction = new FindExpensiveOperationAction();
+		this._register(findExpensiveOperationAction);
+
+		const clearExpensiveOperationAction = new TurnOffExpensiveOperationAction();
+		this._register(clearExpensiveOperationAction);
+
+		const cancelExpensiveOperationAction = new CancelExpensiveOperationAction();
+		this._register(cancelExpensiveOperationAction);
+
+		const self = this;
+		this._operationNameSelectBox.selectElem.onkeydown = async (ev) => {
+			if (ev.key === 'Enter') {
+				if (this._selectedExpensiveOperationType === ExpensiveOperationType.Off) {
+					await clearExpensiveOperationAction.run(self);
+				}
+				else {
+					await findExpensiveOperationAction.run(self);
+				}
+
+				this.showStoreDefaultMetricPrompt();
+			}
+			else if (ev.key === 'Escape') {
+				this.widgetController.removeWidget(self);
+			}
+		};
+
+		const applyButton = new Button(this.container, {
+			title: localize('findExpensiveOperationButtonTitle', 'Find Expensive Operation (Enter)')
```

Comment:
I saw it being used in the `customZoomWidget`, and used it in the expensive node widget, but I will remove them both since I'm also not seeing it being used in other areas in the product.